feat(release): support collecting issues from multiple repos

derekbit opened the pull request:
Introduce owners and repos for release command, so the issues in a release note can be from multiple owner and repo pairs.

innobead commented on src/cmds/release.rs:
```diff
@@ -101,27 +107,54 @@ pub struct ReleaseArgs {
 impl CliCommand for ReleaseArgs {
     async fn run(&self, _cli: &Cli) -> anyhow::Result<()> {
-        let git = GitCli::new(self.owner.clone(), self.repo.clone());
+        if self.ticket_source_owners.is_empty() || self.ticket_source_repos.is_empty() {
+            return Err(anyhow!("ticket_source_owners and ticket_source_repos must be provided"));
+        }
```
This is not required, since these new fields in ReleaseArgs are not optional. Mandatory fields w/ validation already.

derekbit replied:
Addressed. Thank you.

innobead commented on src/cmds/release.rs:
```diff
@@ -31,6 +31,12 @@ pub struct ReleaseArgs {
+    #[arg(long, help = "GitHub owners where tickets are searched. Owners can be multiple owners, separated by colon")]
+    ticket_source_owners: String,
+
+    #[arg(long, help = "GitHub repos where tickets are searched. Repos can be multiple repos, separated by colon")]
+    ticket_source_repos: String,
```
Suggest using https://docs.rs/clap/latest/clap/struct.Arg.html#method.value_delimiter.

```suggestion
    #[arg(long, value_delimiter=",", help = "List of GitHub org/repo where issues are searched")]
    issue_source_owner_repo_pairs: Option<Vec<String>>,
```

Make this field optional, as the default org and repo will be used if not specified.

derekbit replied:
Updated. Thank you.

innobead reviewed (changes requested):
Just a few feedback.

derekbit committed:
feat(release): support collecting issues from multiple repos

Introduce owners and repos for release command, so the issues in a
release note can be from multiple owner and repo pairs.

Signed-off-by: Derek Su <derek.su@suse.com>
